pkg/storj/encryption.go

```diff
@@ -131,6 +131,14 @@ const (
 	NonceSize = 24
 )
 
+// NewKey creates a new key from a passphrase
+func NewKey(passphrase []byte) *Key {
+	// TODO: if/v3-1541 This function should return an error when len(passphrase) == 0
+	key := &Key{}
+	copy(key[:], passphrase)
```

Comment:
Perhaps, it is a good moment to improve this logic here. It's a backward incompatible change, so better do it now than later.

`copy(key[:], passphrase)` has the disadvantage that only the 32 first characters of the passphrase matter. This means that the following two will produce the same encryption key:
- ThisIsARelativelyLongButStillReasonablePassphrase
- ThisIsARelativelyLongButStillReasonableEncryptionKey

It would be better if we take the SHA-256 hash of the passphrase, which will produce `[32]byte`, which is exactly the same size as `storj.Key`.

Reply:
btw, we have a nice `pkcrypto.SHA256Hash` helper function to do this.

Reply:
@jtolds what do you think about this? does it make sense to do it now, or are we going to introduce completely different approach with the design doc for encryption key managment?

Reply:
It makes sense to me, let's see what @jt says about it

Reply:
i definitely thought we were hashing passphrases and we absolutely should for sure. let's do it

Reply:
actually, let's use PBKDF2 instead of sha256 though. that way people can't "rainbow table" a decryption.

Reply:
are you sure we aren't already doing that? i really thought we were doing PBKDF2 or Scrypt or something

Reply:
at the time that I created the PR which was replaced by this one, I am sure, but right now I will have to check with the last upstream updates

Reply:
I discover a few months ago these kind of algorithms for generating keys.

I knew that there was another one called Argon2 and I wanted to know the difference between it and PBKDF2 however I'm no  sure that the sources that I found are reliable but I wanted to mention it here.

Just to mention one of them https://paragonie.com/blog/2016/02/how-safely-store-password-in-2016

Go has the implementation of both 

* Argon2: https://godoc.org/golang.org/x/crypto/argon2
* PBKDF2: https://godoc.org/golang.org/x/crypto/pbkdf2

I'm wondering if there is any special reason to use PBKDF2 rather than Argon2, for example being FIPS compatibility?

Reply:
nope! argon2 seems fine. you pick what you'd like to use

Reply:
I was also pretty sure we were using pbkdf2. Are we really not?

Reply:
@jtolds fine, I already put in place pbkdf2, I was just curious and have the chance to know something new that I didn't know or find. Thanks!

@thepaul I've never seen using any kind of these algorithms in this part of the code, so it may be use but not in this part.